packages/design-system/src/components/Tabs/Tabs.tsx

```diff
@@ -1,23 +1,56 @@
-import PropTypes from 'prop-types';
-import React from 'react';
+/* eslint-disable react/no-multi-comp */
+import React, { useState, useRef } from 'react';
 import Tab from './Tab';
 import TabPanel from './TabPanel';
 import classnames from 'classnames';
 import get from 'lodash/get';
 
+export interface TabsProps {
+  /**
+   * Must only contain `TabPanel` components
+   */
+  children: React.ReactNode;
+  /**
+   * Sets the initial selected `TabPanel` state. If this isn't set, the first
+   * `TabPanel` will be selected.
+   */
+  defaultSelectedId?: string;
+  /**
+   * A callback function that's invoked when the selected tab is changed.
+   * `(selectedId, prevSelectedId) => void`
+   */
+  onChange?: (selectedId: string, prevSelectedId: string) => void;
+  /**
+   * Additional classes to be added to the component wrapping the tabs
+   */
+  tablistClassName?: string;
+}
+
 /** CONSTANTS
  * Adding in the constant values for keycodes
  * to handle onKeyDown events
  */
 const LEFT_ARROW = 'ArrowLeft';
 const RIGHT_ARROW = 'ArrowRight';
 
+/**
+ * Determine if a React component is a TabPanel
+ * @param {React.Node} child - a React component
+ * @return {Boolean} Is this a TabPanel component?
+ */
+function isTabPanel(child): boolean {
+  const componentName = get(child, 'type.displayName') || get(child, 'type.name');
+
+  // Check child.type first and as a fallback, check child.type.displayName follow by child.type.name
+  return child && (child.type === TabPanel || componentName === 'TabPanel');
+}
+
 /**
  * Get the id of the first TabPanel child
  * @param {Object} props
  * @return {String} The id
  */
-function getDefaultSelectedId(props) {
+function getDefaultSelectedId(props): string {
```

Comment:
Similar to above, in this file we mix function syntax, it would be great to agree on one so we can all adopt a pattern for consistency, my vote in the last comment